fails to build pythonPackages.graph-tool

## Issue description

I fail to compile graph-tool on nixos-unstable. 
The error message is a template one, thus 
```
/nix/store/6qmfmricr58bjivgr16y9sjisakdykag-gcc-7.3.0/include/c++/7.3.0/bits/unordered_map.h:976:20:   required from 'std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::mapped_type& std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::operator[](const key_type&) [with _Key = CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<CGAL::Triangulation_data_structure_3<CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<void> >, CGAL::Delaunay_triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_ds_cell_base_3<void> > >, CGAL::Sequential_tag> > >; _Tp = long unsigned int; _Hash = graph_tool::hash_point; _Pred = std::equal_to<CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<CGAL::Triangulation_data_structure_3<CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<void> >, CGAL::Delaunay_triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_ds_cell_base_3<void> > >, CGAL::Sequential_tag> > > >; _Alloc = std::allocator<std::pair<const CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<CGAL::Triangulation_data_structure_3<CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<void> >, CGAL::Delaunay_triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_ds_cell_base_3<void> > >, CGAL::Sequential_tag> > >, long unsigned int> >; std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::mapped_type = long unsigned int; std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::key_type = CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<CGAL::Triangulation_data_structure_3<CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<void> >, CGAL::Delaunay_triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_ds_cell_base_3<void> > >, CGAL::Sequential_tag> > >]'
graph_triangulation.hh:117:23:   required from 'void graph_tool::get_triangulation<Triang, IsPeriodic>::operator()(Graph&, Points&, PosMap) const [with Graph = boost::undirected_adaptor<boost::adj_list<long unsigned int> >; Points = boost::multi_array_ref<double, 2>; PosMap = boost::checked_vector_property_map<std::vector<double>, boost::typed_identity_property_map<long unsigned int> >; Triang = CGAL::Delaunay_triangulation_3<CGAL::Epick>; IsPeriodic = std::integral_constant<bool, false>]'
graph_triangulation.cc:97:42:   required from here
/nix/store/6qmfmricr58bjivgr16y9sjisakdykag-gcc-7.3.0/include/c++/7.3.0/bits/stl_function.h:356:20: error: no match for 'operator==' (operand types are 'const CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<CGAL::Triangulation_data_structure_3<CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<void> >, CGAL::Delaunay_triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_ds_cell_base_3<void> > >, CGAL::Sequential_tag> > >' and 'const CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<CGAL::Triangulation_data_structure_3<CGAL::Triangulation_vertex_base_3<CGAL::Epick, CGAL::Triangulation_ds_vertex_base_3<void> >, CGAL::Delaunay_triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_cell_base_3<CGAL::Epick, CGAL::Triangulation_ds_cell_base_3<void> > >, CGAL::Sequential_tag> > >')
       { return __x == __y; }
                ~~~~^~~~~~
In file included from /nix/store/6qmfmricr58bjivgr16y9sjisakdykag-gcc-7.3.0/include/c++/7.3.0/stack:61:0,
                 from /nix/store/fx9vjsmn753g2ffz4saykg10b3rk90dr-cgal-4.13/include/CGAL/Triangulation_3.h:41,
                 from graph_triangulation.cc:38:
/nix/store/6qmfmricr58bjivgr16y9sjisakdykag-gcc-7.3.0/include/c++/7.3.0/bits/stl_stack.h:293:5: note: candidate: template<class _Tp, class _Seq> bool std::operator==(const std::stack<_Tp, _Seq>&, const std::stack<_Tp, _Seq>&)
     operator==(const stack<_Tp, _Seq>& __x, const stack<_Tp, _Seq>& __y)
     ^~~~~~~~
/nix/store/6qmfmricr58bjivgr16y9sjisakdykag-gcc-7.3.0/include/c++/7.3.0/bits/stl_stack.h:293:5: note:   template argument deduction/substitution failed:
In file included from /nix/store/6qmfmricr58bjivgr16y9sjisakdykag-gcc-7.3.0/include/c++/7.3.0/memory:76:0,
                 from include/boost/config/no_tr1/memory.hpp:21,
                 from include/boost/get_pointer.hpp:14,
                 from include/boost/python/object/pointer_holder.hpp:11,

```

### Steps to reproduce
`nix-build -A pythonPackages.graph-tool channels:nixos-unstable`

cc @joelmo

--- a/pkgs/development/python-modules/graph-tool/2.x.x.nix
+++ b/pkgs/development/python-modules/graph-tool/2.x.x.nix
@@ -1,7 +1,9 @@
-{ stdenv, fetchurl, python, cairomm, sparsehash, pycairo, autoreconfHook,
-pkgconfig, boost, expat, scipy, cgal, gmp, mpfr,
-gobject-introspection, pygobject3, gtk3, matplotlib, ncurses,
-buildPythonPackage }:
+{ stdenv, fetchurl, python, cairomm, sparsehash, pycairo, autoreconfHook
+, pkgconfig, boost, expat, scipy, cgal, gmp, mpfr
+, gobject-introspection, pygobject3, gtk3, matplotlib, ncurses
+, buildPythonPackage
+, fetchpatch
+}:
 
 buildPythonPackage rec {
   pname = "graph-tool";
@@ -19,6 +21,15 @@ buildPythonPackage rec {
     url = "https://downloads.skewed.de/graph-tool/graph-tool-${version}.tar.bz2";
     sha256 = "0w7pd2h8ayr88kjl82c8fdshnk6f3xslc77gy7ma09zkbvf76qnz";
   };
+
+  patches = [
+    # fix build with cgal 4.13 (https://git.skewed.de/count0/graph-tool/issues/509)
+    (fetchpatch {
+      name = "cgal-4.13.patch";
+      url = "https://git.skewed.de/count0/graph-tool/commit/aa39e4a6b42d43fac30c841d176c75aff92cc01a.patch";
+      sha256 = "1578inb4jqwq2fhhwscn5z95nzmaxvmvk30nzs5wirr26iznap4m";
+    })
+  ];
 
   configureFlags = [
     "--with-python-module-path=$(out)/${python.sitePackages}"

